### PR TITLE
Add clustering and association

### PR DESCRIPTION
**Pull request by KilnOfTheSecondFlame**
Damn that took way longer than expected <.<

### CODE REVIEW

File: FS19/ML/Zusammenfassung_ML.tex [hunk 1]
```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
...
+			c_i = \underset{j\in\{1,...,k\}}{\text{argmin}}\ \| x_i - \mu_j \|^2
+		\end{equation*}
+		\item  Update each cluster center to the mean of all assigned data points
+		\begin{equation*}
+			\mu_j := \frac{\sum_{i:c_i=j} x_i}{|\{i:c_i = j\}|}
+		\end{equation*}
+	\end{enumerate}
+\end{enumerate}
+
+\subsubsection{Addendum}
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex:1724]: What is this (OwO)?

File: FS19/ML/Zusammenfassung_ML.tex [hunk 2]
```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
...
+An \textbf{association rule} is an implication $X\rightarrow Y$, where $X$ and $Y$ are disjoint item sets (refer to \ref{fig:associationrule}).
+
+\begin{figure}[htb!]
+	\centering
+	\includegraphics[keepaspectratio, width=0.7\linewidth]{Pictures/association_rule}
+	\caption{A graphical representation of an Association Rule Learner}
+	\label{fig:associationrule}
+\end{figure}
+
+\subsubsection{Support of a Set of Items}
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex:1773]: Why subsubsection instead of just subsection?

File: FS19/ML/Zusammenfassung_ML.tex [hunk 3]
```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
...
+
+For a given rule $X \rightarrow Y$, the higher the confidence, the more likely it is for $Y$ to be present in transactions that contain $X$. Confidence therefore measures reliability or trustworthiness of a rule, and a \textbf{good association rule has high confidence}.
+
+\begin{theorem}
+	Confidence = Trustworthiness
+\end{theorem}
+
+\subsection{Apriori Algorithm}
+
+Given a set of transaction we want to find all rules having a minimum support $\geq \text{min}_s$ and a minimum confidence $\geq \text{min}_c$. The Apriori Algorithm finds such association rules in two steps:
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: *transactions

File: FS19/ML/Zusammenfassung_ML.tex [hunk 4]
```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
...
+\end{enumerate}
+
+\begin{figure}[tbh!]
+	\centering
+	\includegraphics[width=0.5\linewidth, keepaspectratio]{Pictures/superset_pruning}
+	\caption{Pruning of supersets of sets with low support}
+	\label{fig:supersetpruning}
+\end{figure}
+
+After the pruning we generate Rules from the Frequent Item Sets and prune this tree as well. The pruning method for pruning is based on the rule that if a rule $[X \rightarrow Y - X]$ violates the confidence threshold, then any rule $[X’ \rightarrow Y - X’]$ with $X’$ being a subset of $X$ violates the confidence threshold as well (refer to figure \ref{fig:subsetpruning})
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: the pruning method for pruning? Is that supposed to be like that or is there a pruning too many?

**Review by gathurian — CHANGES_REQUESTED** [review]

File: FS19/ML/Zusammenfassung_ML.tex (hunk 1 above)
Comment (KilnOfTheSecondFlame) [reply FS19/ML/Zusammenfassung_ML.tex:1724]: grafik

File: FS19/ML/Zusammenfassung_ML.tex (hunk 2 above)
Comment (KilnOfTheSecondFlame) [reply FS19/ML/Zusammenfassung_ML.tex:1773]: Because I'm stupid

File: FS19/ML/Zusammenfassung_ML.tex (hunk 4 above)
Comment (KilnOfTheSecondFlame) [reply FS19/ML/Zusammenfassung_ML.tex]: We should prune the pruning

**Review by gathurian — APPROVED** [review]

**Pull request merged**